llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2741,6 +2741,17 @@ Instruction *InstCombinerImpl::visitSub(BinaryOperator &I) {
   return TryToNarrowDeduceFlags();
 }
 
+static FastMathFlags combineFastMathFlagsForFNeg(FastMathFlags FMF,
+                                                 FastMathFlags OpFMF) {
+  FastMathFlags Flag = FMF | OpFMF;
+  Flag.setNoInfs(FMF.noInfs() && OpFMF.noInfs());
+  Flag.setAllowReassoc(FMF.allowReassoc() && OpFMF.allowReassoc());
+  Flag.setApproxFunc(FMF.approxFunc() && OpFMF.approxFunc());
+  Flag.setAllowReciprocal(FMF.allowReciprocal() && OpFMF.allowReciprocal());
+  Flag.setAllowContract(FMF.allowContract() && OpFMF.allowContract());
+  return Flag;
+}
```

Comment:
We should have an | value flag & rewrite flag helper in FastMathFlags but can do API cleanups later